include/Board.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include "King.h"
+#include "Bishop.h"
+#include "Queen.h"
+#include "Rook.h"
+#include "Knight.h"
+#include "Pawn.h"
+
+class Board {
+
+public:
+    static const int BOARD_SIZE = 8;
```

Comment:
for constants like this use `constexpr`.
Check it out: https://learn.microsoft.com/en-us/cpp/cpp/constexpr-cpp?view=msvc-170#constexpr-variables